staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -2658,6 +2658,10 @@ const (
 	// disruption (such as preemption, eviction API or garbage-collection).
 	// The constant is to be renamed once the name is accepted within the KEP-3329.
 	AlphaNoCompatGuaranteeDisruptionTarget PodConditionType = "DisruptionTarget"
+	// ResourceExhausted indicates the pod is about to be deleted due to either
+	// exceeding its ephemeral storage limits or running an OOM killed container.
```

Comment:
For now I add it whenever there is a pod failure (and we detect exceeding of disk limits or OOM kill). This still means I don't add the condition in case of OOM killed container and the pod continues to run as the container is restarted.

Reply:
Please add this semantics to the comment.

Reply:
Updated the comment. Two remarks though:
1. the pod may not be necessarily yet in the Failed phase as kubelet makes the transition once all the containers are actually stopped, so typically the condition is added before the failed phase is send to the api-server.
2. After another thought I now add the ResourceExhausted condition only when `spec.restartPolicy=Never`. I think it simplifies the code as it makes the disruption controller change unnecessary. If we know a container exceeded its limits and it is not restarted, then the pod exceeded its limits, and this will not change - so the condition will never be stale. Note that, Job pod failure policy can currently only be used with `restartPolicy=Never` anyway.

If you think 2. makes sense then actually I don't need this new PodCondition type in the API as it will only be used by kubelet. Should we drop this PR then or declare the types still for better visibility? What do you think @liggitt and @alculquicondor ?